File: mobile/library/kotlin/src/io/envoyproxy/envoymobile/ResponseTrailersBuilder.kt
package io.envoyproxy.envoymobile

/*
 * Builder used for constructing instances of `ResponseTrailers`.
 */
class ResponseTrailersBuilder: HeadersBuilder {
  /**
   * Initialize a new instance of the builder.
   */
  constructor() : super(mutableMapOf())

  /**
   * Instantiate a new builder. Used only by ResponseTrailers to convert back to
   * ResponseTrailersBuilder.
   *
   * @param trailers: The trailers to start with.
   */
  internal constructor(trailers: MutableMap<String, MutableList<String>>) : super(trailers)

  /**
   * Build the response trailers using the current builder.
   *
   * @return ResponseTrailers, New instance of response trailers.
   */
  fun build(): ResponseTrailers {
    return ResponseTrailers(headers)
  }
}

File: mobile/library/swift/ResponseTrailersBuilder.swift
import Foundation

/// Builder used for constructing instances of `ResponseTrailers`.
@objcMembers
public final class ResponseTrailersBuilder: HeadersBuilder {
  /// Initialize a new instance of the builder.
  public override convenience init() {
    self.init(container: HeadersContainer())
  }

  /// Build the response trailers using the current builder.
  ///
  /// - returns: New instance of response trailers.
  public func build() -> ResponseTrailers {
    return ResponseTrailers(container: self.container)
  }
}
